src/components/Popup/Popup.js

```diff
@@ -0,0 +1,207 @@
+import React, { useState, useContext, useEffect } from "react";
+import { Link } from "react-router-dom";
+import Backdrop from "../Backdrop/Backdrop";
+import styles from "./Popup.module.scss";
+import cross from "../../images/Cross.svg";
+import crossd from "../../images/CrossD.svg";
+import Catalogue from "../SwapCatalogue/SwapCatalogue";
+import {
+  useAccount,
+  useNetwork,
+  useSwitchNetwork,
+  useProvider,
+  useContract,
+} from "wagmi";
+import { Alchemy, Network } from "alchemy-sdk";
+import axios from "axios";
+import GodwokenNFTs from "../../ABIs/GodwokenNFTs.json";
+import { BsWallet2 } from "react-icons/bs";
+import { ThemeContext } from "../../App";
+import { GODWOKEN_NFTS_ADDRESS } from "../../constants/constants";
+
+function Popup(props) {
+  const [chainConfig, setConfig] = useState(null);
+  const [alchemy, setAlchemy] = useState(null);
+  const [userNFTs, setUserNFTs] = useState([]);
+  const { chain } = useNetwork();
+  const themes = useContext(ThemeContext);
+  const { theme } = themes;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const provider = useProvider();
+  const contract = useContract({
+    addressOrName: GODWOKEN_NFTS_ADDRESS,
+    contractInterface: GodwokenNFTs.abi,
+    signerOrProvider: provider,
+  });
+
+  const { address, connector, isConnected } = useAccount();
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      let config;
+      if (chain) {
+        switch (chain.network) {
+          case "homestead":
+            config = {
+              apiKey: process.env.REACT_APP_ALCHEMY_API_KEY,
+              network: Network.ETH_MAINNET,
+            };
+            break;
+          case "matic":
+            config = {
+              apiKey: process.env.REACT_APP_ALCHEMY_API_KEY,
+              network: Network.MATIC_MAINNET,
+            };
+            break;
+          case "rinkeby":
+            config = {
+              apiKey: process.env.REACT_APP_ALCHEMY_API_KEY,
+              network: Network.ETH_RINKEBY,
+            };
+            break;
+          case "maticmum":
+            config = {
+              apiKey: process.env.REACT_APP_ALCHEMY_API_KEY,
+              network: Network.MATIC_MUMBAI,
+            };
+            break;
+          case "Godwoken ":
```

Comment:
please make sure this extra space after Godwoken won't cause any bugs

Reply:
yeah its working fine